assignments/serializers.py

```diff
@@ -28,16 +30,86 @@ class BudgetingTaskSerializer(serializers.ModelSerializer):
 
     class Meta:
         model = BudgetingTask
-        fields = ['name', 'unit', 'targets', 'amount_of_consumption']
+        fields = ['id', 'name', 'unit', 'targets', 'amount_of_consumption']
+        extra_kwargs = {
+            'name': {'read_only': True},
+            'unit': {'read_only': True},
+            'amount_of_consumption': {'read_only': True}
+        }
 
     def get_unit(self, obj):
         return obj.get_unit_display()
 
 
 class SectionSerializer(serializers.ModelSerializer):
-    opentexttasks = OpenTextTaskSerializer(many=True)
-    budgetingtasks = BudgetingTaskSerializer(many=True)
+    open_text_tasks = OpenTextTaskSerializer(many=True, source='opentexttasks')
+    budgeting_tasks = BudgetingTaskSerializer(many=True, source='budgetingtasks')
 
     class Meta:
         model = Section
-        fields = ['id', 'title', 'description', 'video', 'opentexttasks', 'budgetingtasks']
+        fields = ['id', 'title', 'description', 'video', 'open_text_tasks', 'budgeting_tasks']
+
+
+class OpenTextAnswerSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = OpenTextAnswer
+        fields = ['student', 'task', 'answer']
+        extra_kwargs = {
+            'student': {'read_only': True},
+        }
+
+
+class BudgetingTargetAnswerSerializer(serializers.ModelSerializer):
+    point = serializers.JSONField(required=False, allow_null=True)
+
+    class Meta:
+        model = BudgetingTargetAnswer
+        fields = ['student', 'task', 'target', 'amount', 'point']
+        extra_kwargs = {
+            'student': {'read_only': True}
+        }
+
+    def validate_point(self, data):
+        if not data:
+            return None
+        if type(data) is not list:
+            raise serializers.ValidationError('Invalid point format. You have to specify list')
+        get_json_data = {
+            'type': 'Point',
+            'coordinates': data
+        }
+        return get_json_data
+
+
+class StudentSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Student
+        fields = ['school', 'school_class', 'assignment']
+
+
+class SubmitAnswersSerializer(serializers.Serializer):
+    student = StudentSerializer()
+    open_text_tasks = OpenTextAnswerSerializer(many=True)
+    budgeting_targets = BudgetingTargetAnswerSerializer(many=True)
+
+    def to_internal_value(self, data):
+        """
+        We are adding assignment record to student data.
+        """
+        if 'student' in data:
+            assignment = Assignment.objects.filter(slug=self.context['assignment_slug']).first()
+            if assignment is None:
+                raise serializers.ValidationError(
+                    {'detail': _('Assignment with given slug: {} does not exist').format(self.context['assignment_slug'])})
+            data['student']['assignment'] = assignment.id
```

Comment:
Also, the `detail` key that is used in the value error does not  correspond to a field in the serializer. DRF normally uses `non_field_error` for errors that are not linked to a specific field but to the serializer/form/data in general.